math/classical_sieve_of_eratosthenes.cpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * Classical Sieve of Eratosthenes is an algorithm to find the primes
+ * that is between 2 to N (as defined in main) in fastest way.
+ *
+ * Time Complexity  : O(N * log(log(N)))
+ * Space Complexity : O(N)
+**/
+
+#include <iostream>
+
+bool* isprime = new bool[100000];
```

Comment:
still you are allocating a large amount of memory, let me explain you a bit more clearly

suppose you need to find the prime numbers between 2 and 30, in this case N = 30
so it is sufficient to have an array of size 30 instead of having an array of size 1000000, because
we will use only the first 30 blocks of memory rest all the blocks are useless in this case then there is no need to
take those resources from the system which are not required.
so what we can do in this situation is this we can allocate the memory of 30 blocks by writing
`bool *isPrime = new bool[30];`

and in this particular case you are taking the value of N in the `main()` function so declare this
array of bool in `main()` after taking the value of N, and in order to pass this array of bools to some other functions just pass the pointer to this array by reference to that function, now you don't need to worry to much about how to pass pointer to reference because in C/C++ arrays are automatically passed by reference.

and one last thing is after the program finished then you have to deallocate the resources which you have taken from the system thus it's good to deallocate this memory which you have taken earlier, and to delete this memory write
`delete [] isPrime;`

now it's a  bit more clear let me know if you have any other doubt.